superduperdb/misc/auto_schema.py

```diff
@@ -0,0 +1,117 @@
+import json
+import typing as t
+
+import numpy as np
+
+from superduperdb import logging
+from superduperdb.components.datatype import (
+    DataType,
+    dill_base,
+    json_serializer,
+    serializers,
+)
+from superduperdb.components.schema import Schema
+from superduperdb.ext.numpy.encoder import array
+
+try:
+    import torch
+
+    torch_cls = (torch.Tensor, torch.nn.Module)
+except ImportError:
+    torch_cls = ()
+
+_BASE_TYPES = (
+    int,
+    str,
+    float,
+    bool,
+    bytes,
+    bytearray,
+)
+
+
+def infer_datatype(data: t.Any, ibis=False) -> t.Optional[DataType]:
+    """
+    Infer the datatype of a given data object
+    If the data object is a base type, return None,
+    Otherwise, return the inferred datatype
+
+    :param data: The data object
+    """
+
+    datatype = dill_base
+
+    if isinstance(data, _BASE_TYPES):
+        if not ibis:
+            datatype = None
+        else:
+            from superduperdb.backends.ibis.field_types import dtype
+
+            datatype = dtype(type(data))
+
```

Comment:
Just an idea for scalable solution if we try to add future new datatypes. 

when we create DataType class we can register the datatype in DataType class variable REGISTERED_DATATYPES;

also for each Datatype we can create a type check method which check if data is of `cls` type.

This way we can create something like below:

```python
for type in DataType.registered_types:
    if type.check(data):
       serializers[datatype.identifier] =  type.create(data)
```

Reply:
Cool, Greate Idea.

Reply:
Cool, Great Idea.